app/src/hooks/useCpkAllowance.tsx

```diff
@@ -15,7 +16,7 @@ import { useCpk } from './useCpk'
  */
 export const useCpkAllowance = (signer: Signer, tokenAddress: string) => {
   const cpk = useCpk()
-  const [allowance, setAllowance] = useState<Maybe<BigNumber>>(null)
+  const [allowance, setAllowance] = useState<RemoteData<BigNumber>>(RemoteData.notAsked())
```

Comment:
Converted allowance to a `RemoteData` to have better information about its state.